rep-2014.rst

```diff
@@ -0,0 +1,238 @@
+REP: 2014
+Title: Benchmarking performance in ROS 2
+Author: Víctor Mayoral-Vilches <victor@accelerationrobotics.com>
+Status: Draft
+Type: Informational
+Content-Type: text/x-rst
+Created: 29-Sept-2022
+Post-History: 11-Oct-2022
+
+
+Abstract
+========
+
+This REP describes some principles and guidelines for benchmarking performance in ROS 2.
+
+
+Motivation
+==========
+
+Benchmarking is the act of running a computer program to assess its relative performance. In the context of ROS 2, performance information can help robotists design more efficient robotic systems and select the right hardware for their robotic application. It can also help understand the trade-offs between different algorithms that implement the same capability, and help them choose the best approach for their use case. Performance data can also be used to compare different versions of ROS 2 and to identify regressions. Finally, performance information can be used to help prioritize future development efforts.
+
+
+The myriad combinations of robot hardware and robotics software make assessing robotic-system performance in an architecture-neutral, representative, and reproducible manner challenging. This REP attempts to provide some guidelines to help robotists benchmark their systems in a consistent and reproducible manner by following a quantitative approach. This REP also provides a set of tools and examples to help guide robotists while collecting and reporting performance data.
+
+Value for stakeholders:
+
+- Package maintainers can use these guidelines to integrate performance benchmarking data in their packages.
+
+- Consumers can use the guidelines in the REP to benchmark ROS Nodes and Graphs in an architecture-neutral, representative, and reproducible manner, as well as the corresponding performance data offered in ROS packages to set expectations on the capabilities of each.
+
+- Hardware vendors and robot manufacturers can use these guidelines to show evidence of the performance of their systems solutions with ROS in an architecture-neutral, representative, and reproducible manner.
+
+The guidelines in here are intended to be a living document that can be updated as new information becomes available.
+
+
+A quantitative approach
+-----------------------
+Performance must be studied with real examples and measurements on real robotic computations, rather than simply as a collection of definitions, designs and/or marketing actions. The quantitative approach [1]_ to robotics systems architecture fits well in this context and helps robotic architects come up with better performing architectures through an empirical strategy.
```

Comment:
I like this suggestion. Would you like to make a suggestion out of it @sloretz or should i just pick it?

Reply:
Made an attempt myself to integrate your input @sloretz. Let me know if you still think this needs further improvements:

```suggestion
Performance must be studied with real examples and measurements on real robotic computations, rather than simply as a collection of definitions, designs and/or marketing actions. When creating benchmarks, prefer to use realistic data and situations rather than data designed to test one edge case. Follow the quantitative approach [1]_ when designing your architecture.
```